contracts/protocol/Market/Market.sol

```diff
@@ -3,15 +3,49 @@ pragma solidity ^0.4.11;
 import "./Offer.sol";
 import "../Requirements/Requirements.sol";
 import "../../dependencies/ERC20.sol";
+import "../Profile/Profile.sol";
 
 contract Market{
 
 	Offer[] products;
 	Requirements[] requirements;
 	ERC20 public token;
+	uint public v;
+	mapping (address => Profile) users;	
 
-	function Market(ERC20 _token) {
-		token = _token; 
+	function Market(ERC20 _token, address creator) {
+		token = _token;
+		users[creator] = new Profile();
+	}
+
+	function calculatePrice(Offer _offer, uint _quantity) constant returns (uint) {
+		return _offer.pricePerPackage()*_quantity;
+	}
+
+	function buy(Offer _offer, uint _quantity) {
+		EscrowedAgreement agreement = new EscrowedAgreement(token, _offer, _quantity, msg.sender);
+		assert(token.transferFrom(msg.sender, this, agreement.amount()));
+		if (!token.transfer(agreement, agreement.amount()))
```

Comment:
Why not one transfer instead of two?
`assert(token.transferFrom(msg.sender, agreement, agreement.amount()));`